server/src/test/java/io/crate/executor/transport/task/elasticsearch/DocRefResolverTest.java

```diff
@@ -90,65 +83,4 @@ public void testSystemColumnsCollectExpressions() throws Exception {
         assertThat(collectExpressions.get(5).value()).isEqualTo(1L);
         assertThat(collectExpressions.get(6).value()).isEqualTo(1L);
     }
-
-    // https://github.com/crate/crate/issues/13990
-    @Test
-    public void test_convert_empty_or_null_arrays_added_dynamically_to_nulls() {
-        Doc doc = new Doc(0, "t", "1", 1L, 0L, 1L,
-                          // only useful values here
-                          Map.of("a", 1,
-                              "x", List.of(), // empty array
-                                 "y", Collections.nCopies(3, null), // array of nulls
-                                 "o", Map.of("oo", Map.of("oox", List.of(), // nested empty array
-                                             "ooy", Collections.nCopies(3, null)))), // nested array of nulls
-                          () -> {
-                              throw new AssertionError("this should not be used");
-                          });
-
-        Reference x = new SimpleReference(
-            new ReferenceIdent(
-                new RelationName(null, "doc"),
-                new ColumnIdent("x")),
-            RowGranularity.DOC,
-            DataTypes.STRING, // as long as this is not an array, the test should pass
-            0,
-            null);
-        var collectExpression = REF_RESOLVER.getImplementation(x);
-        collectExpression.setNextRow(doc);
-        assertThat(collectExpression.value()).isNull();
-
-        Reference y = new SimpleReference(
-            new ReferenceIdent(
-                new RelationName(null, "doc"),
-                new ColumnIdent("y")),
-            RowGranularity.DOC,
-            DataTypes.UNTYPED_OBJECT, // as long as this is not an array, the test should pass
-            0,
-            null);
-        collectExpression = REF_RESOLVER.getImplementation(y);
-        collectExpression.setNextRow(doc);
-        assertThat(collectExpression.value()).isNull();
```

Comment:
Fails with Cast expression. This is because it doesn't really correspond to production code.

We don't have [direct source access](https://github.com/crate/crate/commit/75cce24bc142da15a6239b4d39113f249ab8d77a#diff-d07f183da84b28eb1023b661fb71485dc40297c1d541ab5bb470d5acf430ed81L109) anymore for PKLookup, it's already processed by `SourceParser` which handles all this.

I could fix this by adding SourceParser to the map but feel like it's redundant and betetr to remove the test.
Adding more components to the unit tests to align it with actual code looks redundant as we already have integration tests for coverage and a